rocksdb_admin/admin_handler.cpp

```diff
@@ -249,6 +249,73 @@ std::unique_ptr<rocksdb::DB> AdminHandler::removeDB(
   return db;
 }
 
+void AdminHandler::async_tm_addDB(
+      std::unique_ptr<apache::thrift::HandlerCallback<std::unique_ptr<
+          AddDBResponse>>> callback,
+      std::unique_ptr<AddDBRequest> request) {
+  AdminException e;
```

Comment:
can you add the following code at the beginning of this function.

`db_admin_lock_.Lock(request->db_name);
SCOPE_EXIT { db_admin_lock_.Unlock(request->db_name); };
`

It helps us to synchronize all DB admin operations.